### PR TITLE
Set Provider Type

### PR DESCRIPTION
**Pull request by analoo**
#### 🔗 Jira ticket
No Jira ticket. 

#### ✍️ Description
- The intent of this ticket is to set provider type so that in the later screens in the process, we can conditionally show screens to some provider types but not others.
- Because the provider type logic was moved to the SetProviderType action, and the value is stored in provider type, we can simplify the preparer code a lot.

#### ✅ Completion tasks

- [ ] Added relevant tests
- [ ] Meets acceptance criteria

### CODE REVIEW

File: src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        } else if ("false".equals(providerInputData.get("providerCurrentlyLicensed"))) {
+            submission.getInputData().put("providerType", unLicensedProviderType(providerInputData));
+        }
+    }
+
+    private String licensedProviderType(Map<String, Object> inputData) {
+        String providerLicensedCareLocation = (String) inputData.getOrDefault("providerLicensedCareLocation", "");
+        switch (providerLicensedCareLocation) {
+            case "childCareCenter":
+                return ProviderType.LICENSED_DAY_CARE_HOME.name();
```
Comment (enyia21) [inline src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java:29]:
```suggestion
                return ProviderType.LICENSED_DAY_CARE_CENTER.name();
              
```

Comment (enyia21) [reply src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java:29]: I think this error is causing the test to fail

File: src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                if (providerLicenseExemptRelationship.equals("Relative")) {
+                    return ProviderType.LICENSE_EXEMPT_RELATIVE_IN_PROVIDER_HOME.name();
+                } else if (providerLicenseExemptRelationship.equals("Not related")) {
+                    return ProviderType.LICENSE_EXEMPT_NONRELATIVE_IN_PROVIDER_HOME.name();
+                }
+            } else if ("Childs home".equals(inputData.get("providerLicenseExemptCareLocation"))) {
+                if (providerLicenseExemptRelationship.equals("Relative")) {
+                    return ProviderType.LICENSE_EXEMPT_RELATIVE_IN_CHILDS_HOME.name();
+                } else if (providerLicenseExemptRelationship.equals("Not related")) {
+                    return ProviderType.LICENSE_EXEMPT_RELATIVE_IN_CHILDS_HOME.name();
```
Comment (enyia21) [inline src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java:56]:
```suggestion
                    return ProviderType.LICENSE_EXEMPT_NONRELATIVE_IN_CHILDS_HOME.name();
```

**Review by enyia21 — CHANGES_REQUESTED** [review]

File: src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java (hunk 2 above)
Comment (analoo) [reply src/main/java/org/ilgcc/app/submission/actions/SetProviderType.java:56]: Thank you @enyia21

**Review by cram-cfa — APPROVED** [review]

**Review by enyia21 — APPROVED** [review]

**Pull request merged**